src/j2735-pcap/decodeJ2735.py

```diff
@@ -0,0 +1,132 @@
+    f = open(inputFile, 'r')
+    Lines = f.readlines()
+    f.close()
+
+    return Lines
+
+def writeIds(w, msgId_count):
+    w.write('\nDecoded Message ID Counts:\n')
+    print('\nDecoded Message ID Counts:')
+    for msgId, count in msgId_count.items():
+        w.write(f'{msgId}: {count}\n')
+        print(f'{msgId}: {count}')
+
+def isValidMessage(line):
+    tempFrame = line[6:]
+    if (len(tempFrame.strip('\n')) > 510):
+        frameSize = 8
+        encodedSize = int(line[5:8], 16) * 2
+    else: 
+        frameSize = 6
+        encodedSize = int(line[4:6], 16) * 2
+
+    newFrame = line[frameSize:].strip('\n')
+    if (encodedSize == len(newFrame)):
+        return True
+    else:
+        return False
+    
+def convertBytes(obj):
+    if isinstance(obj, dict):
+        return {k: convertBytes(v) for k, v in obj.items()}
+    elif isinstance(obj, list):
+        return [convertBytes(item) for item in obj]
+    elif isinstance(obj, tuple):
+        return [convertBytes(item) for item in obj]
+    elif isinstance(obj, bytes):
+        return obj.hex()
+    else:
+        return obj
+
+def decode(data, frame, w, msgId_count, id, csv_output, i):
+    #Write data to text file
+    w.write(data)
+    w.write('\n')
+
+    frame.from_uper(unhexlify(data))
+    cleanObj = convertBytes(frame())
+    jsonString = json.dumps(cleanObj, indent=2)
+    jsonDict = json.loads(jsonString)
+    message_type = jsonDict['value'][0]
+    data_value = data
+    #Write data to the CSV file
+    csv_writer = csv.writer(csv_output)
+    packet_number = i
+    packet_count = capture[i]
+    timestamp = packet_count.sniff_time
+    if i==1:
+        # Write the CSV header
+        csv_writer.writerow(['Packet Number', 'Timestamp', 'Message Type', 'Payload'])
+    csv_writer.writerow([packet_number, timestamp, message_type, data_value])
+
+    #Write json string to text file
+    w.write(jsonString)
+    w.write('\n')
+
+    msgId_count[id] += 1  # increment count for successfully decoded msgId
+
+def main():
+    frame = J2735_201603_2023_06_22.DSRC.MessageFrame
+    msgIds = ['0012','0013','0014','001f','0020','0029'] # can be updated to include other PSIDs
+    msgId_count = defaultdict(int)  # dictionary to track decoded msgId and their counts
+    fileName = formatFileName()
+    fileName_csv = formatFileName_csv()
+    w = open(decodedDir + '/' + fileName, 'w')
+    csv_output = open(decodedDir + '/' + fileName_csv, 'w', newline='')
+    i=1 # Iterator for packet number
+    for line in readLines():
+        for id in msgIds:
+            idx = line.find(id)
+            if (idx != -1):
+                data = line[idx:].strip('\n')
+                if (isValidMessage(data) == True):
+                    try:
+                        decode(data, frame, w, msgId_count, id, csv_output, i)
+                        i += 1
+                    except Exception as e:
+                        print("Error decoding message: ", e)
```

Comment:
You are not capturing any information about the packets you fail to decode. Please capture at least a count of failures and print it our at the end.
```python
print(f'Decoding of file {} Complete.  Successfully decoded {} msgs, Failed to decode {}, ' fileName, successCount, errorCount,)

```